base/reflection.jl

```diff
@@ -343,7 +343,10 @@ If `x === y` then `objectid(x) == objectid(y)`, and usually when `x !== y`, `obj
 
 See also [`hash`](@ref), [`IdDict`](@ref).
 """
-objectid(@nospecialize(x)) = ccall(:jl_object_id, UInt, (Any,), x)
+function objectid(@nospecialize(x))
```

Comment:
Is this true? 
```
const A = zeros(3)
f() = objectid(A)
```

Will have a foldable objectid during a session, but not across sessions.

```
julia> using ObjId

julia> ObjId.f()
0xb541a7170b98f837
```

Next session:
```
julia> ObjId.greet()
0x1a54cacafa58e6f3
```

So does foldable imply "we can write this result into a cache file"?

Reply:
good point. It's ugly to fix, but we can't quite do this.